spec/docx/smoke/api_paragraph_spec.rb

```diff
@@ -269,4 +269,72 @@
     expect(json['type']).to eq('paragraph')
     expect(json['content'][1]['content'][0]).to eq('This is a paragraph')
   end
+
+  describe 'ApiParagraph | AddCaption method' do
+    before(:all) do
+      @docx = builder.build_and_parse('js/docx/smoke/api_paragraph/add_caption.js')
```

Comment:
When using instance variable in the `before` block, we solve the problem of multiple execution of resource-intensive code. But at the same time, we may encounter an unpleasant side effect.
The initialized `instance variable` refers to the area in memory where the object is written and this object is available for overwriting. A scenario is possible where for the first `it` object in `state 1` it will be executed correctly and for the second block `it` instance variable will refer to the already modified `object` under test.

A possible solution would be to `freeze` the object being passed to instance variable
https://rubyapi.org/3.3/o/object#method-i-freeze